GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -55,33 +69,20 @@ public void Dispose()
             }
         }
 
-        private static EventMetadata CreateEventMetadata(Exception e)
-        {
-            EventMetadata metadata = new EventMetadata();
-            if (e != null)
-            {
-                metadata.Add("Exception", e.ToString());
-            }
-
-            return metadata;
-        }
-
-        private void TimerCallback(object unusedState)
+        public void TimerCallback(object unusedState)
         {
             string errorMessage = null;
 
             using (ITracer activity = this.tracer.StartActivity("Checking for product upgrades.", EventLevel.Informational))
             {
                 try
                 {
-                    ProductUpgraderInfo info = new ProductUpgraderInfo(
-                        this.tracer,
-                        this.fileSystem);
-
                     // The upgrade check always goes against GitHub

```

Comment:
This would change to checking against well known server with anonymous access, right? Can we test this (which server gets used for upgrade checks)?